prac_09/os_demos.py

```diff
@@ -0,0 +1,61 @@
+"""
+CP1404/CP5632 Practical
+Demos of various os module examples
+"""
+import shutil
+import os
+
+
+def main():
+    """Demo os module functions."""
+    print("Starting directory is: {}".format(os.getcwd()))
+
+    # Change to desired directory
+    os.chdir('Lyrics/Christmas')
+
+    # Print a list of all files in current directory
+    print("Files in {}:\n{}\n".format(os.getcwd(), os.listdir('.')))
+
+    try:
+        os.mkdir('temp')
+    except FileExistsError:
+        pass
+
+    # Loop through each file in the (current) directory
+    for filename in os.listdir('.'):
+        # Ignore directories, just process files
+        if os.path.isdir(filename):
+            continue
+
+        new_name = get_fixed_filename(filename)
+        print("Renaming {} to {}".format(filename, new_name))
+
+        # Option 1: rename file to new name - in place
+        os.rename(filename, new_name)
+
+        # Option 2: move file to new place, with new name
+        shutil.move(filename, 'temp/' + new_name)
+
+
+def get_fixed_filename(filename):
+    """Return a 'fixed' version of filename."""
+    new_name = filename.replace(" ", "_").replace(".TXT", ".txt")
+    return new_name
+
+
+def demo_walk():
+    """Process all subdirectories using os.walk()."""
+    os.chdir('Lyrics')
```

Comment:
Looking good. Only thing I can think of is to possibly have a constant or something similar for this Lyrics folder and possibly also for the above file link but it probably isn't needed